File: LeetCode/code/1.两数之和/TwoSum.cpp
#include <iostream>
#include <vector>
using namespace std;
class solution{
    public:
    vector<int> twosum(vector<int>& nums,int target){
        int i,j;
        for(int i=0;i<nums.size()-1;i++){
            for(int j=i+1;j<nums.size();j++){
                if(nums[i]+nums[j]==target){
                    return vector<int>{i,j};
                }
            }
        }
        return vector<int>();
    };
};
int main(){
    cout<<"请输入数组元素个数：\n";
    int n;
    cin>>n;
    vector<int> arr(n);// 使用vector定义数组，根据输入个数动态分配内存
     cout<<"请输入元素：\n";
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
    int target;
    cout<<"请输入目标值：\n";
    cin>>target;
    solution s;
    vector<int> result=s.twosum(arr,target);

    if(result.size()==2){
        cout<<"结果为：\n";
        cout<<"["<<result[0]<<","<<result[1]<<"]"<<endl;
    }
    else{
        cout<<"没有找到符合条件的结果"<<endl;
    }
}

File: LeetCode/code/1.两数之和/TwoSum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i+1,len(nums)):
                if nums[i]+nums[j]==target:
                    return [i,j]
                return []
arr = []
input_str = input("输入数组元素（空格分隔开，回车结束）:")
elements = input_str.split()
for element in elements:
    arr.append(int(element))

target = int(input("请输入目标值："))
s=Solution()
result=s.twoSum(arr,target)
if result:
    print("找到两个数的索引为：",result)
else:
    print("未找到两个数")